src/app/auth.provider.js

```diff
@@ -0,0 +1,139 @@
+        getAuthInfo: getAuthInfo
+      };
+
+      function login(options) {
+        var d = $q.defer();
+        clbAppHello.login('hbp', options)
+        .then(function(res) {
+          d.resolve(getAuthInfo(res.authResponse));
+        }, function(err) {
+          d.reject(_formatError(err));
+        });
+        return d.promise;
+      }
+
+      function logout(options) {
+        var info = getAuthInfo();
+        if (!info) {
+          return $q.when(true);
+        }
+        var d = $q.defer();
+        clbAppHello.logout('hbp', options)
+        .then(function() {
+          return d.resolve(true);
+        }, function(err) {
+          d.reject(_formatError(err));
+        });
+        return d.promise;
+      }
+
+      function getAuthInfo(authResponse) {
+        authResponse = authResponse || clbAppHello.getAuthResponse('hbp');
+        if (!authResponse) {
+          return null;
+        }
+        return {
+          accessToken: authResponse.access_token,
+          tokenType: authResponse.token_type,
+          // When no scopes are specified, the server will generate a token
+          // with the app default scopes. In this case hello.js don't know what
+          // they are so we set the value to undefined by convention.
+          scope: authResponse.scope || undefined,
+          expires: authResponse.expires
+        };
+      }
+
+      function _formatError(err) {
+        return clbError.error({
+          type: err.error.code,
+          message: err.error.message,
+          data: err
+        });
+      }
+
+      function _bindEvents() {
+        clbAppHello.on('auth.login', _handleAuthInfoChange);
+        clbAppHello.on('auth.logout', _handleAuthInfoChange);
+      }
+
+      function _handleAuthInfoChange(data, name) {
+        if (data.network !== 'hbp') {
+          return;
+        }
+        $log.debug('propagate auth event from original event', name);
+        $timeout(function() {
+          $rootScope.$broadcast('clbAuth.changed', getAuthInfo());
+        }, 0);
+      }
+
+      /**
+       * Define a new provider Hello.js provider for HBP
+       */
+      function _addHbpProvider() {
+        clbAppHello.init({
+          hbp: {
+            name: 'Human Brain Project',
+            oauth: {
+              version: '2',
+              auth: clbEnvProvider.get('auth.url') + '/authorize',
+              grant: clbEnvProvider.get('auth.url') + '/token'
+            },
+            // API base URL
+            base: clbEnvProvider.get('auth.url') + '/',
+            scope_delim: ' ', // eslint-disable-line camelcase
+            login: function(p) {
+              // Reauthenticate
+              if (p.options.force) {
+                p.qs.prompt = 'login';
+              }
+              if (!p.qs.scope) {
+                delete p.qs.scope;
+              }
+            },
+            logout: function(callback, p) {
+              $http.post(clbEnv.get('auth.url') + '/slo', {
```

Comment:
@zanitete, This call return a 200 but then it seems I am still able to log in without my credentials being asked for (because the OIDC session is still there probably). I think this is broken, right?